dwave/cloud/utils.py

```diff
@@ -303,6 +304,74 @@ def create_url(self, url):
         return urljoin(self.base_url, url)
 
 
+def hasinstance(iterable, class_or_tuple):
+    """Extension of ``isinstance`` to iterables/sequences. Returns True iff the
+    sequence contains at least one :class:func`isinstance` of ``class_or_tuple``.
+    """
+
+    return any(isinstance(e, class_or_tuple) for e in iterable)
+
+
+def exception_chain(exception):
+    """Traverse the chain of embedded exceptions, yielding one at the time.
+
+    Args:
+        exception (:class:`Exception`): Chained exception.
+
+    Yields:
+        :class:`Exception`: The next exception in the input exception's chain.
+
+    Examples:
+        def f():
+            try:
+                1/0
+            except ZeroDivisionError:
+                raise ValueError
+
+        try:
+            f()
+        except Exception as e:
+            assert(hasinstance(exception_chain(e), ZeroDivisionError))
+
+    See: PEP-3134.
+    """
+
+    while exception:
+        yield exception
+
+        # explicit exception chaining, i.e `raise .. from ..`
+        if exception.__cause__:
+            exception = exception.__cause__
+
+        # implicit exception chaining
+        elif exception.__context__:
+            exception = exception.__context__
+
+        else:
+            return
+
+
+def hasexception(exception, exception_types):
```

Comment:
I think this is the point where consistency with `isinstance` is less compelling than readability.  ` hasinstance` work but this one initially parses for me as "ha sex ception". Perhaps give it a second thought before going ahead.